Services/CO.CDP.DataSharing.WebApi/UseCase/GenerateShareCodeUseCase.cs

```diff
@@ -1,16 +1,27 @@
 using AutoMapper;
+using CO.CDP.Authentication;
 using CO.CDP.DataSharing.WebApi.Extensions;
 using CO.CDP.DataSharing.WebApi.Model;
 using CO.CDP.OrganisationInformation.Persistence;
 using CO.CDP.OrganisationInformation.Persistence.Forms;
 
 namespace CO.CDP.DataSharing.WebApi.UseCase;
 
-public class GenerateShareCodeUseCase(IFormRepository formRepository, IMapper mapper)
+public class GenerateShareCodeUseCase(
+    IClaimService claimService,
+    IOrganisationRepository organisationRepository,
+    IFormRepository formRepository,
+    IMapper mapper)
     : IUseCase<ShareRequest, ShareReceipt>
 {
     public async Task<ShareReceipt> Execute(ShareRequest shareRequest)
     {
+        var org = await organisationRepository.Find(shareRequest.OrganisationId);
+        if (org == null || claimService.GetOrganisationId() != org.Id)
+        {
+            throw new InvalidOrganisationRequestedException("Invalid Organisation requested.");
```

Comment:
Is there a plan to unique share codes?

Reply:
Discussed offline with Gavin, part of previous PR, agreed to mark as resolved.